sandy/core/cov.py

```diff
@@ -800,6 +811,11 @@ def sampling(self, nsmp, seed=None, rows=None, pdf='normal',
         # pdf
         if pdf == 'uniform':
             to_decompose = self.__class__(np.diag(np.diag(self.data)))
+            if cutzeros:
+                a = np.sqrt(12) / 2
+                std = pd.DataFrame(np.sqrt(np.diag(np.diag(self.data))))

```

Comment:
minor comment, but i don't like calling this `std` since technically it is a matrix with the standard deviations on the diagonal.

Second comment: I would do
```
std = pd.DataFrame(np.sqrt(np.diag(np.diag(to_decompose.data))))
```

Third comment: independently from the second comment, since this matrix only contains non-zero values on the diagonal, isn't it an overkill to look for a condition over all elements of the matrix? if it is for simplicity of implementation I would understand.
But, wouldn't it be simpler to do:
```
std = np.sqrt(np.diag(to_decompose.data))
std_modified = np.where(std < a,  std, a)
cov = np.diag(std.modified)
to_decompose = self.__class__(cov, index=to_decompose.data.index, columns=to_decompose.datacolumns)
```
Maybe not, since at the end you need to reinitialize the dataframe with index and columns.

Up to you.
Let me know